### PR TITLE
Comments: `CommentDetail` Redux refactor

### PR DESCRIPTION
**Pull request by Copons**
- Moved all props to the components who actually use it.
- Created an util file for recurring tasks.
- Removed `CommentDetail` from the devdocs, since it's now completely dependent from the `CommentList`.

No existing logic has been changed or harmed, this PR really just needs a smoke test on the comments section.

Next task: move `CommentDetail` away from `/blocks` and into `/my-sites/comment-list`.

Another task could be to re-nest some children objects, such as `author` or `post`, that are usually needed in their entirety (or close to that).

### CODE REVIEW

File: client/blocks/comment-detail/comment-detail-actions.jsx [hunk 1]
```diff
@@ -123,4 +125,13 @@ export const CommentDetailActions = ( {
 	);
 };
 
-export default localize( CommentDetailActions );
+const mapStateToProps = ( state, { commentId, siteId } ) => {
+	const comment = getSiteComment( state, siteId, commentId );
+
+	return {
+		commentIsLiked: get( comment, 'i_like' ),
+		commentStatus: get( comment, 'status' ),
```
Comment (gwwar) [inline client/blocks/comment-detail/comment-detail-actions.jsx]: I'm a little confused on why we pass though ever other prop but these two.

**Comment by gwwar** [comment]
So as an overall comment let's be careful to pick which components are pure UI pieces and which ones are blocks. The former only cares about content display, and things like analytics handlers are passed through to onClick from their parents. With the right props we can render whatever we need. Blocks generally declare their data dependencies, so they are droppable and reusable in any context.

If we feel like we're repeating ourselves, that may be an opportunity to create a small block component.

I think some of the pain is being caused by the number of props we need off of comment. I'd also be open to passing that down a single level, provided we clearly document the shape of the comment that needs to be passed.

**Comment by dmsnell** [comment]
I didn't look at the code, but nice LOC delta

**Comment by Copons** [comment]
@rodrigoi I've rebased this PR, but I think we should put it on hold until both #18657 and #18624 are merged to avoid propagating conflicts all over the place.

**Comment by Copons** [comment]
Replaced by full redesign (see https://github.com/Automattic/wp-calypso/pull/19082).

**Pull request closed without merging**